File: broker/src/main/java/lambdamart/service/broker/graphql/model/Vendor.java
package lambdamart.service.broker.graphql.model;

public class Vendor {
    private String title;
    private String description;
    private String icon;
    private InventoryItem[] inventory;

    public Vendor(String title, String description, String icon, InventoryItem[] inventory) {
        this.title = title;
        this.description = description;
        this.icon = icon;
        this.inventory = inventory;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public InventoryItem[] getInventory() {
        return inventory;
    }

    public void setInventory(InventoryItem[] inventory) {
        this.inventory = inventory;
    }
}

File: lambda-market/service/src/schema/vendor.ts
import { gql } from "graphql-request";

export const typeDef = gql`
  # Represents a vendor that sells products through the broker
  type Vendor {
    title: String!
    description: String!
    icon: String!
    inventory: [InventoryItem!]!
  }

  # Represents a vendors inventory of a single item
  type InventoryItem {
    id: ID!
    stockLevel: Int!
    price: Float!
  }
`;
